test/k8sT/Services.go

```diff
@@ -1599,17 +1599,18 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`, helpers.DualStackSupp
 				// but not from k8s2 to k8s1. In the k8s2 to k8s1 case, kube-proxy
 				// would send traffic to k8s1, where it would be subsequently
 				// dropped, because k8s1 has no service backend.
-				// However, if HostReachableServices is enabled, then Cilium does
-				// the service translation already on the client node, bypassing
-				// kube-proxy completely.
+				// If HostReachableServices is enabled, Cilium does the service
+				// translation for ClusterIP services on the client node, bypassing
+				// kube-proxy completely. Here, we are probing NodePort service, so we
+				// need BPF NodePort to be enabled as well for the requests to succeed.
 				hostReachableServicesTCP := kubectl.HasHostReachableServices(ciliumPodK8s2, true, false)
 				hostReachableServicesUDP := kubectl.HasHostReachableServices(ciliumPodK8s2, false, true)
-				if hostReachableServicesTCP {
+				if hostReachableServicesTCP && helpers.RunsWithoutKubeProxy() || !isSupported(k8sVersion) {
```

Comment:
The comment at L1593 needs to be updated. 
>we are probing NodePort service, so we
// need BPF NodePort to be enabled as well for the requests to succeed.

Where are we checking if the node port config is enabled?

Reply:
It's ugly, but currently the only way to check if BPF NodePort is enabled is to check `helpers.RunsWithoutKubeProxy()`, as done here. It relies on the fact that our CI pipelines are either kube-proxy free (so BPF NodePort has to be enabled) or running on old kernels with kube-proxy (so BPF NodePort has to be disabled).

#14543 will fix this by introducing a `RunsWithKubeProxyReplacement`. `RunsWithKubeProxy` will check if kube-proxy is installed and not if it is actually used.

Reply:
>currently the only way to check if BPF NodePort is enabled is to check helpers.RunsWithoutKubeProxy()

Can we check if the feature is enabled similar to how we check for [HostReachableServices](https://github.com/cilium/cilium/blob/master/test/helpers/utils.go#L590)?

Reply:
That's a good idea! I updated the branch to do just that.
It won't generalize to #14543 however, because it only works if you already have a Cilium pod up.